### ISSUE TITLE
About to be a Contribution

### ISSUE THREAD

**Issue by dutchsing009** [mindslab-ai/nuwave#19]
I decided to make a colab , and it took hours to adjust the packaging of pip and all the other mismatching stuff , and the colab is supposedly working , and no packaging errors or (Module not found) errors anymore ,, but the thing is , it doesn't really upscale , i used the x3 last.ckpt from freds0 it should upscale 16khz to 48khz , and i did put a 16khz file which was provided from the demo and it did the following
![Annotation 2022-06-08 032011](https://user-images.githubusercontent.com/103085811/172520581-335a5bb4-97bb-4428-8382-327ad96e725d.png)

by the way I don't understand why it did so many outputs , but i took the "sample_0_48000.wav" and there is no upscaling improvement at all , at all :( but the interesting thing is it does say 48khz if imported to any audio software but it doesn't sound like it nor the spectrogram looks like it ,, you know the 16khz file that i experimented on was 66kb , after supposedly the upscaling is made it says it is 197kb but it doesn't sound like 48khz at ALL ,,, here comes the crazy part when i exported the same 16khz 66kb file to adobe audition and then exported it to 48khz which is wrong but i did it , it was the same size as the nuwave one 197 kb also , and it had the same spectrogram of the nuwave ,, then i knew that there is something wrong and that nu wave isn't really functioning alright ,, P.S during the inference there was also some errors but it didn't stop it ,,
the code was; !python sampling.py -c last.ckpt -f sample_0.wav 
and it did that /usr/local/lib/python3.7/dist-packages/torch/functional.py:472: UserWarning: stft will soon require the return_complex parameter be given for real inputs, and will further require that return_complex=True in a future PyTorch release. (Triggered internally at  /pytorch/aten/src/ATen/native/SpectralOps.cpp:664.)
  normalized, onesided, return_complex)
/usr/local/lib/python3.7/dist-packages/torch/functional.py:546: UserWarning: istft will require a complex-valued input tensor in a future PyTorch release. Matching the output from stft with return_complex=True.  (Triggered internally at  /pytorch/aten/src/ATen/native/SpectralOps.cpp:817.)
  normalized, onesided, length, return_complex) 

i hope you reach as soon as possible as this problem of it saying it is 48khz and not sounding like one is really making my mind go crazy :)

Have a nice day/night

**Comment by freds0** [comment]
Sorry, I trained the model but I ran out of time and didn't even get to test the checkpoints. Can you share jupyter notebook? For x2 the model worked?

**Comment by dutchsing009** [comment]
oh yeah of course but as you see above i only tried the x3 one which supposedly should do from 16khz to 48khz , give me an email to send it on ,,

**Comment by freds0** [comment]
@dutchsing009 fred.santos.oliveira@gmail.com

**Comment by dutchsing009** [comment]
> @dutchsing009 [fred.santos.oliveira@gmail.com](mailto:fred.santos.oliveira@gmail.com)

ok i did send the email , you should have it by now :)

**Comment by ghost** [comment]
Do you guys have a working public version for this?

**Comment by dutchsing009** [comment]
Don't stress , it is not really useful , Actually it doesn't do the results you see in the demo at all , as the authors never released the original checkpoints , the checkpoints proposed by Mr fred is good but needs much training as by now it is filled by much noise and it is not satisfactory , i will try to train them more soon , then we will see what happens.

**Comment by ghost** [comment]
Right! Feel free to mention if you put that to work

**Comment by junjun3518** [comment]
@dutchsing009 Hi! Sorry for the late response.
To begin with, it is true that our model is not generally acting for every inputs. In addition, it need to train for a long time. Last year, there are some problems for releasing its checkpoint, but may be we could release it on this year.
In addition, return_complex=False is need for STFT if you use higher version of torch than our code. 
https://pytorch.org/docs/stable/generated/torch.stft.html
Over 1.8.0, return_complex option is required, but lower version, its default value was False.

**Comment by dutchsing009** [comment]
@junjun3518 Hi , No problems at all , a late response is better than no response :) ,,  I saw and read the new amazing paper which is called Nu-Wave 2 , it is far more better and its outputs can be indistinguishable from the reference signals , And also the fact it has the ability to upsample audio signals of any sampling rate such as 11.025 kHz ,, 22.05 kHz , 32 kHz , 44.1 kHz. Please focus on releasing the official checkpoints of Nu-wave 2 as soon as the code is released that would be so amazing ,and that also would compensate the huge delay of the checkpoints of nu-wave 1.

**Comment by psp0001060** [comment]
> I want to experiment with colab,Can you share jupyter notebook?

@dutchsing009
